### PR TITLE
Add track placement to sorting hat

### PR DESCRIPTION
**Pull request by NorthIsUp**
example output in slack https://danceaholics.slack.com/files/adam/F3JK5DE68/example_output_of_track_placement.js

### CODE REVIEW

File: lindy/audition/management/commands/sorting_hat.py [hunk 1]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+        level_quotas = {
+            9: 22 * 2,
+            4: 35 * 2,
+            8: 35 * 2,
+            7: 35 * 2,
+            6: 35 * 2,
+            5: 35 * 2,
+            3: 35 * 2,
+            2: 23 * 2,
+            # 1: 15 * 2,  # handled specially
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: why is level 1 handled specially? You can audition and still end up in this group.

Comment (NorthIsUp) [inline lindy/audition/management/commands/sorting_hat.py]: Because level one is not a part of the Ben distribution of students.

File: lindy/audition/management/commands/sorting_hat.py [hunk 2]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+        min_role_level_quotas = self.make_levels(min_role_ratio, level_quotas_scaled, 'min scaled')
+
+        # sort the roles into tracks
+        max_track_placements = self.sort_into_track(max_role, max_role_level_quotas, name='max')
+        min_track_placements = self.sort_into_track(min_role, min_role_level_quotas, name='min')
+
+        # for logging purposes
+        summation = {k: int(max_role_level_quotas[k] + min_role_level_quotas[k]) for k in level_quotas}
+
+        # log stuff
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: THe logging is just for you, right?  So the fact that what you're logging doesn't make sense to me doesn't matter?

File: lindy/audition/management/commands/sorting_hat.py [hunk 3]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+        self.stderr.write('=  sum {} {}'.format(summation, sum(summation.values())))
+        self.stderr.write('<= scl {} {}'.format(valmap(int, level_quotas_scaled), sum(level_quotas_scaled.values())))
+        self.stderr.write('<= qot {} {}'.format(level_quotas, sum(level_quotas.values())))
+
+        # save the results
+        self.save_placements(max_track_placements, commit=commit)
+        self.save_placements(min_track_placements, commit=commit)
+        self.save_placements({1: level_one}, commit=commit)
+
+    def make_levels(self, ratio, d, name=None):
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: This isn't super important since this code won't be long lived, but calling it something other than "d" would help me understand what the heck it is you're expecting.

File: lindy/audition/management/commands/sorting_hat.py [hunk 4]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+        for the sort_into_tracks method
+        Note: uses fractions so math doesn't mess up
+        '''
+
+        if name:
+            self.stderr.write('make level for {}'.format(name))
+        levels = {}
+        remainder = Fraction.from_decimal(0)
+
+        for level, v in sorted(d.items(), reverse=True):
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: is v the max number of people that can be in this level? Am I reading that right?
Why is it sorted? I don't understand.

Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: it'd be cool if you can tell me (through comments or whatever) what this whole for loop is doing-- hard to follow.

File: lindy/audition/management/commands/sorting_hat.py [hunk 5]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+
+        """
+        if name:
+            self.stderr.write('placing for {}'.format(name))
+        track_placements = {}
+
+        i = 0
+        for level, size in sorted(level_quotas.items(), reverse=True):
+            # from 9 to 1
+            track_placements[level] = auditions[i:size + i]
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: Can you tell me what's happening here?

File: lindy/audition/management/commands/sorting_hat.py [hunk 6]
```diff
@@ -106,3 +166,189 @@ def get_heats(role, heat_count=4):
...
+            i += size
+            self.stderr.write('{} {}'.format(
+                level,
+                ', '.join('{:0.2}'.format(_.average_placement) for _ in track_placements[level])
+            ))
+
+        return track_placements
```
Comment (NicoleZuckerman) [inline lindy/audition/management/commands/sorting_hat.py]: question: 
We should err on the side of putting people in the lower levels a bit, as in not maxing out the upper categories right away.  We're not doing that here, right?

**Comment by NicoleZuckerman** [comment]
@NorthIsUp do you need to land this?

**Comment by NorthIsUp** [comment]
dun
